octue/django-gcp
# Cloud tasks

# Summary


# Contents ([#16](https://github.com/octue/django-gcp/pull/16))

### New features
- Add Cloud Tasks manager, views, urls, Task ABCs and examples

### Enhancements
- Export all members
- Include error handling on requests
- Improve urls definition in example server

### Fixes
- Make message parsers consume full pub/sub messages

### Operations
- Use WP long goldfish stack for development
- Name the postgres volume for clarity, prepare move to test server
- Allow customisation of devcontainer per-developer, add gcloud SDK
- Update devcontainer to latest stack
- Add a terraform config (WIP) for the example server
- Bump version for new feature release
- Remove broken extensions for restructuredText
- Add localtunnel install to the dockerfile
- Update grpcio to non-yaked version
- Make grpcio maintainers suffer for yanking a key release
- Use psycopg2-binary to ensure pg_config available on GHA

### Dependencies
- Resolve bad merge
- Widen django dependencies to include 4.x

### Refactoring
- Separate the example app from teh tests directory
- Make error name clearer
- Remove unused routing file
- Remove unused asgi file
- Add an OnDemandTask class to avoid future breaking change

### Testing
- Fix broken test caused by adding redirect
- Make storage tests work with example assets
- Add test suite up to 90% coverage
- Patch the gcp authorisation

### Other
- Add instructions for using the test app
- Add a redirect to the root in the example app
- Use terraform-built assets in example server
- Merge main
- Update docs and include Tasks
- Expand temporary note about where to find how-to example
- Added watcher to replace broken extensions
- Added a little extra to teh tasks usage docs
